back/app/services/survey_results_generator_service.rb

```diff
@@ -384,7 +390,7 @@ def group_query(query, group: false)
   end
 
   def generate_answer_keys(field)
-    (field.input_type == 'linear_scale' ? (1..field.maximum).reverse_each.to_a : field.options.map(&:key)) + [nil]
+    (field.input_type.in?(%w[linear_scale rating]) ? (1..field.maximum).to_a : field.options.map(&:key)) + [nil]
```

Comment:
Maybe use `%w[linear_scale rating].include?(field.input_type)` instead as this tends to be the pattern we use